settlement/src/execution.rs

```diff
@@ -50,7 +50,11 @@ pub fn create_execution_transaction(
             format!("ex-transfer-nft: {}", execution.target_chain)
         }
     };
-    let body = serde_spb::to_string(&execution).unwrap();
+    let body = format!(
+        "{}\n---\n{}",
+        serde_spb::to_string(&execution).unwrap(),
```

Comment:
Using a method like "split" is expensive in solidity. How about concatenating hash first?